docs/runtimes/functions.md

```diff
@@ -1,27 +1,73 @@
 (Functions)=
-# Functions
+# Functions 
 
-All the executions in MLRun are based on **Serverless functions**. The functions allow specifying code and 
-all the operational aspects (image, required packages, [cpu/mem/gpu resources](../runtimes/configuring-job-resources.html#cpu-gpu-and-memory-limits-for-user-jobs), [storage](../runtimes/function-storage.html), environment, etc.). 
-The [different function runtimes](../concepts/functions-overview.html) take care of automatically transforming the code and spec to fully 
-managed and elastic services over Kubernetes, which saves significant operational overhead, 
-addresses scalability and reduces infrastructure costs.
+All the executions in MLRun are based on serverless functions. Functions are essentially Python code that can be 
+executed locally or on a Kubernetes cluster. 
+MLRun functions are used to run jobs, deploy models, create pipelines, and more. 
+
+There are various kinds of MLRun functions with different capabilities, however, 
+there are commonalities across all functions. In general, an MLRun function looks like the following:
 
-MLRun supports:
-- Real-time functions for: {ref}`serving <serving-graph>`, APIs, and stream processing (based on the high-performance {ref}`Nuclio <nuclio-real-time-functions>` engine). 
-- Batch functions (based on Kubernetes jobs, {ref}`Spark <spark-operator>`, {ref}`Dask <dask-overview>`, {ref}`Horovod <horovod>`, etc.)
+<img src="../_static/images/mlrun-functions.png" alt="mlrun-architecture" width="700"/><br>
+
+MLRun supports numerous real-time and batch runtimes, described in {ref}`function_runtimes`. 
+The different function runtimes take care of automatically transforming the code and spec to fully 
+managed and elastic services over Kubernetes, which saves significant operational overhead, 
+addresses scalability, and reduces infrastructure costs. The function parameters and capabilities 
+are explained in more detail in {ref}`create-and-use-functions`.
 
-Function objects are all inclusive (code, spec, API, and metadata definitions), which allows placing them 
-in a shared and versioned function market place. This means that different members of the team can produce or 
-consume functions. Each function is versioned and stored in the MLRun database with a unique hash code, 
+Function objects are all inclusive, containing the code and all the operational aspects: ({ref}`image <images-usage>`, required packages, 
+{ref}`pod resource congifuration (replicas, CPU/GPU/memory limits, volumes, Spot vs. On-demand nodes, pod priority, node affinity) <configuring-job-resources>`, 
+[storage](../runtimes/function-storage.html), environment, metadata definitions, etc.). Each function is versioned and stored in the MLRun 
+database with a unique hash code, 
 and gets a new hash code upon changes.
 
+MLRun functionality such as logging results and artifacts can be added to existing code using the {py:func}`mlrun.handler` decorator, 
+enabled by default. 
```

Comment:
Since the handler decorator is now enabled by default (as part of packagers) I would say it's actually a bad practice to use the handler anymore. Instead we should provide details on how to specify these configurations using the `returns` parameter and the input type hints. @guy1992l , @alonmr  - WDYT?

Reply:
Agreed

Reply:
@theSaarco  - updated text, pls check